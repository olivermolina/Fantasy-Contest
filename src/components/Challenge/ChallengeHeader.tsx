import React, { useState } from 'react';
import StarRateIcon from '@mui/icons-material/StarRate';
import { HowToPlayDialog } from './HowToPlay';
import classNames from 'classnames';
import Skeleton from '@mui/material/Skeleton';

const styles = {
  container:
    'flex flex-col lg:flex-row lg:justify-between text-white py-4 px-4 bg-primary min-h-[100px] gap-4',
  button: 'text-white normal-case font-semibold',
  divider: 'h-1 w-1 rounded-full bg-white',
  promoContainer:
    'flex flex-grow justify-center items-center bg-secondary rounded-lg p-4 gap-2',
  linksContainer: 'flex flex-row gap-5 justify-around items-center',
};

interface Props {
  /** The message to display in the promo container */
  challengePromoMessage: string;
  /** Whether the component is loading */
  isLoading: boolean;
  /** The function to call when the contact us button is clicked */
  handleOpenContactUs: () => void;
}

export default function ChallengeHeader(props: Props) {
  const [openHowToPlay, setOpenHowToPlay] = useState(false);
  const handleCloseHowToPlay = () => {
    setOpenHowToPlay(false);
  };

  const handleOpenHowToPlay = () => {
    setOpenHowToPlay(true);
  };

  return (
    <div className={styles.container}>
      <div className={styles.promoContainer}>
        <StarRateIcon fontSize="small" />
        {props.isLoading ? (
          <Skeleton
            variant={'text'}
            width={'50%'}
            sx={{ bgcolor: '#144e97' }}
          />
        ) : (
          props.challengePromoMessage
        )}
      </div>
      <div className={styles.linksContainer}>
        <button className={styles.button} onClick={props.handleOpenContactUs}>
          Contact Us
        </button>
        <span className={styles.divider} />
        <button className={styles.button} onClick={handleOpenHowToPlay}>
          How To Play
        </button>
        <span className={classNames(styles.divider, 'hidden')} />
        <button className={classNames(styles.button, 'hidden')}>
          Scoring Chart
        </button>
      </div>
      <HowToPlayDialog
        open={openHowToPlay}
        handleClose={handleCloseHowToPlay}
      />
    </div>
  );
}
