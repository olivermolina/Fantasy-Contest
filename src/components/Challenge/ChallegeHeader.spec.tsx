import React from 'react';
import { render } from '@testing-library/react';
import ChallengeHeader from './ChallengeHeader';

describe('ChallengeHeader', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ChallengeHeader
        challengePromoMessage={
          'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.'
        }
        isLoading={false}
        handleOpenContactUs={jest.fn()}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
