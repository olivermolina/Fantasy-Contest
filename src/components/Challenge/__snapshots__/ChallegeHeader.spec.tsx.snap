// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeHeader should render correctly 1`] = `
<div>
  <div
    class="flex flex-col lg:flex-row lg:justify-between text-white py-4 px-4 bg-primary min-h-[100px] gap-4"
  >
    <div
      class="flex flex-grow justify-center items-center bg-secondary rounded-lg p-4 gap-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
        data-testid="StarRateIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M14.43 10 12 2l-2.43 8H2l6.18 4.41L5.83 22 12 17.31 18.18 22l-2.35-7.59L22 10z"
        />
      </svg>
      Pick 2-4 players. Predict if they will get MORE or LESS than their projection.
    </div>
    <div
      class="flex flex-row gap-5 justify-around items-center"
    >
      <button
        class="text-white normal-case font-semibold"
      >
        Contact Us
      </button>
      <span
        class="h-1 w-1 rounded-full bg-white"
      />
      <button
        class="text-white normal-case font-semibold"
      >
        How To Play
      </button>
      <span
        class="h-1 w-1 rounded-full bg-white hidden"
      />
      <button
        class="text-white normal-case font-semibold hidden"
      >
        Scoring Chart
      </button>
    </div>
  </div>
</div>
`;
